internal/kuberesource/parts.go

```diff
@@ -31,9 +33,21 @@ type NodeInstallerConfig struct {
 }
 
 // NodeInstaller constructs a node installer daemon set.
-func NodeInstaller(namespace string) *NodeInstallerConfig {
+func NodeInstaller(namespace string, platform platforms.Platform) (*NodeInstallerConfig, error) {
 	name := "contrast-node-installer"
 
+	var podRole, imageURL string
+	switch platform {
+	case platforms.AKSCloudHypervisorSNP:
+		podRole = "contrast-node-installer-microsoft"
```

Comment:
I still think we should split this, keep the role as is and instead introduce an additional platform annotation.